lib/Common/Codex/Utf8Helper.h

```diff
@@ -83,52 +91,82 @@ namespace utf8
 
     inline HRESULT WideStringToNarrowDynamic(LPCWSTR sourceString, LPSTR* destStringPtr)
     {
-        return WideStringToNarrow<malloc_allocator>(sourceString, destStringPtr);
+        size_t unused;
+        return WideStringToNarrow<malloc_allocator>(
+            sourceString, wcslen(sourceString), destStringPtr, &unused);
     }
 
     inline HRESULT NarrowStringToWideDynamic(LPCSTR sourceString, LPWSTR* destStringPtr)
     {
-        return NarrowStringToWide<malloc_allocator>(sourceString, destStringPtr);
+        size_t unused;
+        return NarrowStringToWide<malloc_allocator>(
+            sourceString, strlen(sourceString), destStringPtr, &unused);
     }
 
 
     template <class Allocator, class SrcType, class DstType>
     class NarrowWideStringConverter
     {
     public:
-        static HRESULT Convert(SrcType src, DstType* dst);
+        static size_t Length(const SrcType& src);
+        static HRESULT Convert(
+            SrcType src, size_t srcCount, DstType* dst, size_t* dstCount);
     };
 
     template <class Allocator>
     class NarrowWideStringConverter<Allocator, LPCSTR, LPWSTR>
     {
     public:
-        static HRESULT Convert(LPCSTR sourceString, LPWSTR* destStringPtr)
+        static size_t Length(LPCSTR src)
         {
-            return NarrowStringToWide<Allocator>(sourceString, destStringPtr);
+            return strlen(src);
+        }
+
+        static HRESULT Convert(
+            LPCSTR sourceString, size_t sourceCount,
+            LPWSTR* destStringPtr, size_t* destCount)
+        {
+            return NarrowStringToWide<Allocator>(
+                sourceString, sourceCount, destStringPtr, destCount);
         }
     };
 
     template <class Allocator>
     class NarrowWideStringConverter<Allocator, LPCWSTR, LPSTR>
     {
     public:
-        static HRESULT Convert(LPCWSTR sourceString, LPSTR* destStringPtr)
+        static size_t Length(LPCWSTR src)
+        {
```

Comment:
Don't know what todo you meant exactly. We have wcslen calls elsewhere. I'll add a comment -- Here it is used rarely; Typically we'd ask caller to provide input string length. This is a fallback.